File: backend/apps/clinicas/formas_pago.py
from django.db import OperationalError, ProgrammingError, transaction

FORMAS_PAGO_DEFAULT = [
    # (nombre, tipo_base, orden)
    ("Efectivo", "efectivo", 1),
    ("Transferencia", "transferencia", 2),
    ("Tarjeta débito", "tarjeta_debito", 3),
    ("Tarjeta crédito", "tarjeta_credito", 4),
    ("Crédito", "credito", 5),
    ("Cuotas", "cuotas", 6),
    ("Financiamiento", "financiamiento", 7),
    ("Otro", "otro", 8),
]


def ensure_default_formas_pago_for_clinica(clinica):
    """Siembra el catálogo base de formas de pago de una clínica.

    Idempotente (``update_or_create`` por nombre): sirve tanto para la
    creación de una clínica nueva como para backfill de una existente.
    """
    from apps.clinicas.models import FormaDePago

    try:
        with transaction.atomic():
            for nombre, tipo_base, orden in FORMAS_PAGO_DEFAULT:
                FormaDePago.objects.update_or_create(
                    clinica=clinica,
                    nombre=nombre,
                    defaults={
                        "tipo_base": tipo_base,
                        "es_sistema": True,
                        "activo": True,
                        "orden": orden,
                    },
                )
    except (OperationalError, ProgrammingError):
        return


def resolver_forma_pago_por_tipo_base(clinica, tipo_base):
    """Busca la forma de pago sembrada de ``clinica`` para ``tipo_base``.

    Usado por integraciones que solo conocen el tipo base (p. ej. el
    asistente de puesta en marcha, que recibe un string plano del frontend)
    en vez del id de la fila. Siempre debería existir porque el seed cubre
    todos los ``tipo_base`` — si no, cae a "Otro".
    """
    from apps.clinicas.models import FormaDePago

    return (
        FormaDePago.objects.filter(clinica=clinica, tipo_base=tipo_base, activo=True).first()
        or FormaDePago.objects.filter(clinica=clinica, tipo_base=FormaDePago.TipoBase.OTRO).first()
    )

File: frontend/src/types/formasPago.ts
export type TipoBaseFormaPago =
  | 'efectivo'
  | 'transferencia'
  | 'tarjeta_debito'
  | 'tarjeta_credito'
  | 'credito'
  | 'cuotas'
  | 'financiamiento'
  | 'otro'

export interface FormaDePago {
  id: string
  clinica: string
  nombre: string
  tipo_base: TipoBaseFormaPago
  tipo_base_display: string
  es_sistema: boolean
  activo: boolean
  orden: number
  created_at: string
}

export interface CreateFormaDePagoRequest {
  nombre: string
  tipo_base: TipoBaseFormaPago
  orden?: number
  activo?: boolean
}

export interface UpdateFormaDePagoRequest {
  nombre?: string
  orden?: number
  activo?: boolean
}

/** tipo_base que no son un medio real de cobro (son plan/estructura, no un pago recibido). */
export const TIPOS_BASE_NO_MEDIO_REAL: readonly TipoBaseFormaPago[] = ['cuotas', 'financiamiento']
